### PR TITLE
Update Kotlin, dependencies and build scripts

### PR DESCRIPTION
**Pull request by terrakok**

### CODE REVIEW

File: components/resources/demo/shared/build.gradle.kts [hunk 1]
```diff
@@ -1,13 +1,21 @@
...
-version = "1.0-SNAPSHOT"
-
 kotlin {
-    android()
+    @OptIn(ExperimentalKotlinGradlePluginApi::class)
+    targetHierarchy.default()
+    androidTarget {
+        compilations.all {
+            kotlinOptions {
+                jvmTarget = "11"
```
Comment (MatkovIvan) [inline components/resources/demo/shared/build.gradle.kts:15]: Shouldn't we use `jvmToolchain(11)` instead?

Comment (terrakok) [reply components/resources/demo/shared/build.gradle.kts:15]: the jvmToolchain is not my favorite thing to use. we want to set java version only for the android target. so it seems easier to use explicit jvmTarget.

File: components/gradle.properties [hunk 2]
```diff
@@ -1,21 +1,26 @@
...
+
+#Compose
 org.jetbrains.compose.experimental.jscanvas.enabled=true
 org.jetbrains.compose.experimental.macos.enabled=true
 org.jetbrains.compose.experimental.uikit.enabled=true
+compose.desktop.verbose=true
 
+#Kotlin
+kotlin.code.style=official
+kotlin.mpp.enableCInteropCommonization=true
```
Comment (igordmn) [inline components/gradle.properties:22]: We have a different properties now. Were each property changed consciously?

**Review by igordmn — APPROVED** [review]

File: components/gradle.properties (hunk 2 above)
Comment (terrakok) [reply components/gradle.properties:22]: Yes. I set the properties by intention and removed outdated ones

Comment (terrakok) [reply components/gradle.properties:22]: By fact, it is required only for an IDE code completion. So, I'll roll back it

Comment (dima-avdeev-jb) [reply components/gradle.properties:22]: As I understand this property also affects publication process.
But as I correctly understand previous problems was with SkikoUIView class compatibility.
So, for now - it should be fine!

Comment (igordmn) [reply components/gradle.properties:22]: > it is required only for an IDE code completion

> was with SkikoUIView class compatibility.

Let's keep the property only if we are sure if we need it

**Review by dima-avdeev-jb — APPROVED** [review]

**Pull request merged**